homeworks/03-data-structure/implementations/binary-tree.ts

```diff
@@ -0,0 +1,79 @@
+import {IBinaryTree} from "../models/binary-tree";
+import {TraverseType} from "../models/traverse-type";
+import {TreeNode} from "../models/tree-node";
+
+export class BinaryTree<T> implements IBinaryTree<T> {
+
+    constructor(protected root: TreeNode<T>) {
+    }
+
+    setTree(tree: TreeNode<T>): this {
+        this.root = tree;
+        return this;
+    }
+
+    traverse(traverseType: TraverseType): T[] {
+        if (traverseType === TraverseType.BFS) return this.bfs([this.root], []);
+
+        return this.dfs(this.root, traverseType);
+    }
+
+    getColumn(columnOrder: number): T[] {
+        return this.combineByLevel(columnOrder, 0, this.root);
+    }
+
+    insert(value: T, current?: TreeNode<T>): BinaryTree<T> {
+        const startNode = current || this.root;
+
+        if (value < startNode.value) {
+            if (startNode.left === null) {
+                startNode.left = new TreeNode<T>(value);
+                return this;
+            } else {
+                return this.insert(value, startNode.left);
+            }
+        } else if (value > startNode.value) {
+            if (startNode.right === null) {
+                startNode.right = new TreeNode<T>(value);
+                return this;
+            } else {
+                return this.insert(value, startNode.right);
+            }
+        }
+    };
+
+    private bfs(traverseQueue: TreeNode<T>[], acc: T[]): T[] {
+        if (!traverseQueue.length) return acc;
+
+        const currentNode = traverseQueue.shift();
+        const newQueue = [...traverseQueue, currentNode.left, currentNode.right].filter(value => !!value)
+        return this.bfs(newQueue, [...acc, currentNode.value]);
+    }
+
+    private dfs(node: TreeNode<T> | null, traverseType: TraverseType): T[] {
+        if (!node) return [];
+
+        const left = this.dfs(node.left, traverseType);
+        const right = this.dfs(node.right, traverseType);
```

Comment:
Somewhere you write `;` at the end of the line, somewhere not. You should be consistent.
Use eslint.